src/test/java/com/intel/gkl/pdhmm/IntelPDHMMUnitTest.java

```diff
@@ -0,0 +1,231 @@
+		final boolean isloaded = new IntelPDHMM().load(null);
+
+		final IntelPDHMM intelPDHMM = new IntelPDHMM();
+		Assert.assertTrue(isloaded);
+
+		try {
+			FileInputStream fis = new FileInputStream(pdhmmData);
+			BufferedLineReader br = new BufferedLineReader(fis);
+			br.readLine(); // skip first line
+			int testcase = 0;
+			String line;
+			while ((line = br.readLine()) != null) {
+				testcase++;
+			}
+			br.close();
+
+			int hapArraySize = testcase * HAPLOTYPE_MAX_LENGTH;
+			int readArraySize = testcase * READ_MAX_LENGTH;
+
+			byte[] alleleBasesFull = new byte[hapArraySize];
+			byte[] allelePDBasesFull = new byte[hapArraySize];
+			byte[] readBasesFull = new byte[readArraySize];
+			byte[] readQualsFull = new byte[readArraySize];
+			byte[] readInsQualsFull = new byte[readArraySize];
+			byte[] readDelQualsFull = new byte[readArraySize];
+			byte[] overallGCPFull = new byte[readArraySize];
+			double[] expectedFull = new double[testcase];
+			long[] hapLength = new long[testcase];
+			long[] readLength = new long[testcase];
+
+			// todo - do we need to close and recreate fis ?
+			fis.close();
+			fis = new FileInputStream(pdhmmData);
+			br = new BufferedLineReader(fis);
+			br.readLine(); // skip first line
+
+			int currentTestcase = 0;
+			while ((line = br.readLine()) != null) {
+				String[] split = line.split("\t"); // Assuming the integers are space-separated
+				byte[] alleleBases = split[0].getBytes(StandardCharsets.UTF_8);
+				byte[] allelePDBases = ArrayUtils.toPrimitive(
+						Arrays.stream(split[1].substring(1, split[1].length() - 1).split(","))
+								.map(num -> Byte.parseByte(num.trim())).toArray(Byte[]::new));
+				byte[] readBases = split[2].getBytes(StandardCharsets.UTF_8);
+				byte[] readQuals = SAMUtils.fastqToPhred(split[3]);
+				byte[] readInsQuals = SAMUtils.fastqToPhred(split[4]);
+				byte[] readDelQuals = SAMUtils.fastqToPhred(split[5]);
+				byte[] overallGCP = SAMUtils.fastqToPhred(split[6]);
+				double expected = Double.parseDouble(split[7]);
+
+				// append testcase to full arrays
+				appendArray(alleleBasesFull, alleleBases, currentTestcase * HAPLOTYPE_MAX_LENGTH);
+				appendArray(allelePDBasesFull, allelePDBases, currentTestcase * HAPLOTYPE_MAX_LENGTH);
+				appendArray(readBasesFull, readBases, currentTestcase * READ_MAX_LENGTH);
+				appendArray(readQualsFull, readQuals, currentTestcase * READ_MAX_LENGTH);
+				appendArray(readInsQualsFull, readInsQuals, currentTestcase * READ_MAX_LENGTH);
+				appendArray(readDelQualsFull, readDelQuals, currentTestcase * READ_MAX_LENGTH);
+				appendArray(overallGCPFull, overallGCP, currentTestcase * READ_MAX_LENGTH);
+				expectedFull[currentTestcase] = expected;
+				hapLength[currentTestcase] = alleleBases.length;
+				readLength[currentTestcase] = readBases.length;
+				currentTestcase++;
+			}
+			br.close();
+
+			// Call Function
+			long start = System.nanoTime();
+			double actual[] = intelPDHMM.computePDHMM(alleleBasesFull, allelePDBasesFull,
```

Comment:
Replace with Java-style declaration.
```suggestion
			double[] actual = intelPDHMM.computePDHMM(alleleBasesFull, allelePDBasesFull,
```